libr/core/file.c

```diff
@@ -122,6 +122,10 @@ R_API int r_core_file_reopen(RCore *core, const char *args, int perm, int loadbi
 		// XXX - select the right backend
 		if (core->file && core->file->desc) {
 			newpid = core->file->desc->fd;
+#if __linux__
```

Comment:
main_pid is only being set in linux_debug.c, so we shouldn't have the problem on *BSD.